hw3/hw3-react/src/components/HouseChart.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Doughnut } from 'react-chartjs-2';
+import {
+	Chart as ChartJS,
+	ArcElement,
+	Tooltip,
+	Legend,
+	Colors,
+} from 'chart.js';
+import axios from 'axios';
+
+ChartJS.register(ArcElement, Tooltip, Legend, Colors);
+// url for the Thrones API
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+const backgroundColors = [
+	'rgba(54, 162, 235, 0.8)',
+	'rgba(255, 206, 86, 0.8)',
+	'rgba(255, 99, 132, 0.8)',
+	'rgba(75, 192, 192, 0.8)',
+	'rgba(153, 102, 255, 0.8)',
+	'rgba(255, 159, 64, 0.8)',
+	'rgba(199, 199, 199, 0.8)',
+	'rgba(83, 102, 255, 0.8)',
+	'rgba(40, 159, 64, 0.8)',
+	'rgba(210, 199, 199, 0.8)',
+	'rgba(78, 52, 199, 0.8)',
+];
+
+const borderColors = [
+	'rgba(54, 162, 235, 1)',
+	'rgba(255, 206, 86, 1)',
+	'rgba(255, 99, 132, 1)',
+	'rgba(75, 192, 192, 1)',
+	'rgba(153, 102, 255, 1)',
+	'rgba(255, 159, 64, 1)',
+	'rgba(159, 159, 159, 1)',
+	'rgba(83, 102, 255, 1)',
+	'rgba(40, 159, 64, 1)',
+	'rgba(210, 199, 199, 1)',
+	'rgba(78, 52, 199, 1)',
+];
+const HouseChart = () => {
+	const [data, setData] = useState({});
+	useEffect(() => {
+		const init = async () => {
+			const characters = await fetchData();
+			const characterCounts = countCharactersByHouse(characters);
+			console.log('COUNT', characterCounts);
+			const chartData = renderChart(characterCounts);
+			setData(chartData);
+		};
+		init();
+	}, []);
+
+	const fetchData = async () => {
+		const response = await axios(url);
+		return response.data;
+	};
```

Comment:
Per my previous comment, you would pass the API data to this component so that you wouldn't need this chunk of code that makes the API call since you would already have it.